source/channel.c

```diff
@@ -828,7 +828,7 @@ static void s_window_update_task(struct aws_channel_task *channel_task, void *ar
 
     channel->window_update_scheduled = false;
 
-    if (status == AWS_TASK_STATUS_RUN_READY && channel->channel_state < AWS_CHANNEL_SHUTTING_DOWN) {
+    if (status == AWS_TASK_STATUS_RUN_READY && channel->channel_state < AWS_CHANNEL_SHUT_DOWN) {
```

Comment:
this change needs to happen, but it opens up a can of worms where a handler might get a window-increment callback after it's finished shutting down in the read-dir. So it might naively try to read again and send more read-messages, which is a crime after shutdown.

Can you look into this?

ughghghgh it makes me want to rewrite the channel API so that things like this are impossible. Where the channel tracks basic state information about each handler and won't allow stuff like this (don't alert a handler about window-update if it's already completed shutdown-in-read-dir). So every handler doesn't need to be so paranoid and defensive to be "correct". We don't have time for now right now, this fix needs to get out. But if there are followup issues...